tests/integration/dropdown/test_option_title_prop.py

```diff
@@ -0,0 +1,57 @@
+# -*- coding: UTF-8 -*-
+
+import pytest
+import dash
+from dash.dependencies import Input, Output
+import dash_core_components as dcc
+import dash_html_components as html
+
+from utils import wait_for
+
+
+@pytest.mark.DCC793
+@pytest.mark.parametrize("multi", [True, False])
+def test_ddot001_option_title(dash_dcc, multi):
+    app = dash.Dash(__name__)
+    app.layout = html.Div([
+        dcc.Input(
+            id="dropdown_title_input",
+            type="text",
+            placeholder="Enter a title for New York City"
+        ),
+        dcc.Dropdown(
+            id="dropdown",
+            options=[
+                {'label': 'New York City', 'value': 'NYC'},
+                {'label': 'Montréal', 'value': 'MTL'},
+                {'label': 'San Francisco', 'value': 'SF'}
+            ],
+            value='NYC',
+            multi=multi
+        )
+    ])
+
+    @app.callback(Output("dropdown", "options"), [Input("dropdown_title_input", "value")])
+    def add_title_to_option(title):
+        return [
+            {'label': 'New York City', 'title': title, 'value': 'NYC'},
+            {'label': 'Montréal', 'value': 'MTL'},
+            {'label': 'San Francisco', 'value': 'SF'}
+        ]
+
+    dash_dcc.start_server(app)
+
+    dropdown_option_element = dash_dcc.wait_for_element("#dropdown .Select-value")
+
+    dropdown_title_input = dash_dcc.wait_for_element("#dropdown_title_input")
+
+    dropdown_title_input.send_keys("The Big Apple")
+
+    wait_for(
+        lambda: dropdown_option_element.get_attribute("title")
+        == "The Big Apple",
+        lambda: '`title` is {}, expected {}'.format(
+            dropdown_option_element.get_attribute("title"),
+            "The Big Apple",
+        ),
+    )
```

Comment:
@wbrgss I'm not sure why there's a callback to update the `title` on the 1st option: If all we want to do is make sure it's present, we don't need to. If we want to make sure it's present after an update, we should also test that the `title` attribute is missing or empty prior to the change, to be consistent.

I'm ok with us not testing further (eg. `title` present in the dropdown, not just the selected value) as that would be the same as testing the 3rd party component that we're wrapping.

---

Using `wait_for_element` instead of `find_element_by_css_selector` makes for a slightly more verbose test but also for a slightly more resilient test (eg. timing issues in CI). That said the test app is pretty small and probably will rarely be an issue. Same goes for `wait_for` vs. simply getting the attribute. This is 👌

Reply:
> If we want to make sure it's present after an update, we should change it more than once and make sure it changes None --> "Big Apple" --> "Knicks Game" --> None for example.

Agreed, that makes sense. I'll update it.

> makes for a slightly more verbose test but also for a slightly more resilient test

Yeah, I've had timing issues here and there with more complicated test apps so I've been using `wait_for` for overkill to be safe.

Reply:
>  If we want to make sure it's present after an update, we should change it more than once and make sure it changes None --> "Big Apple" --> "Knicks Game" --> None for example.

@Marc-Andre-Rivet I've done this in d5bee5d; if that looks right to you I'm going to merge.